src/read.c

```diff
@@ -3922,17 +4006,14 @@ avifResult avifDecoderOutputDecodedTiles(avifDecoder * decoder, const avifTileIn
         // Nothing changed.
         return AVIF_RESULT_OK;
     }
-    // There is at least one newly decoded tile.
-    assert(info->decodedTileCount > oldDecodedTileCount);
-
-    // Grid path.
+    // For grid images, the output image is populated in avifDecoderDecodeTiles.
```

Comment:
IMPORTANT: The DecodeTiles() and OutputDecodedTiles() functions no longer do what their names imply after this change. This should be a TODO for future cleanup. It seems that these two functions should be merged.

Reply:
Added TODO In line 4090. I will look into this in a follow-up.